File: ABC/ABC040/B/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);

    int n = sc.nextInt();

    sc.close();

    int min = Integer.MAX_VALUE;

    for (int i = 1; i < 1000; i++) {
      for (int j = 1; j < 1000; j++) {
        int r = n - i * j;
        if (0 <= r) {
          int result = r + Math.abs(i - j);
          min = Math.min(min, result);
        }
      }
    }

    System.out.println(min);

  }

}

File: ABC/ABC040/B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n;
  cin >> n;

  int ans = 999999;

  for (int i = 1; i <= n; i++) {
    int a1 = n / i;
    int a2 = n % i;
    ans = min(ans, abs(a1 - i) + a2);
  }

  cout << ans << endl;

}